src/display/base.jl

```diff
@@ -101,9 +101,15 @@ import Base.Docs: doc
 isanon(f) = startswith(string(typeof(f).name.mt.name), "#")
 
 @render Inline f::Function begin
+  name = string(typeof(f).name)
+  inner = match(r"typeof\((.+)\)", name)
+  if inner ≠ nothing
+    name = inner[1]
+  end
+  binding = Docs.Binding(typeof(f).name.module, Symbol(name))
   isanon(f) ? span(".syntax--support.syntax--function", "λ") :
-    LazyTree(span(".syntax--support.syntax--function", string(typeof(f).name.mt.name)),
-             ()->[(Atom.CodeTools.hasdoc(f) ? [md_hlines(doc(f))] : [])..., methods(f)])
+    LazyTree(span(".syntax--support.syntax--function", name),
+             ()->[(CodeTools.hasdoc(binding) ? [md_hlines(doc(binding))] : [])..., methods(f)])
```

Comment:
so we need to special case macros since `string(typeof(f).name)` would look weird in those cases:

This snippet would address them:
```julia
@render Inline f::Function begin
  name = string(typeof(f).name)
  inner = match(r"typeof\((.+)\)", name)
  if inner ≠ nothing
    name = inner[1]
  end
  (ism = ismacro(f)) && (name = string(f))
  binding = Docs.Binding(typeof(f).name.module, Symbol(name))
  isanon(f) ?
    span(".syntax--support.syntax--function", "λ") :
    LazyTree(
      span(ism ? ".syntax--string.syntax--quote" : ".syntax--support.syntax--function", name),
      ()->[(CodeTools.hasdoc(binding) ? [md_hlines(doc(binding))] : [])..., methods(f)]
    )
end
```